### PR TITLE
querying for unstarted tasks

### PR DESCRIPTION
**Pull request by stan-dot**

### CODE REVIEW

**Comment by stan-dot** [comment]
forgot to update the Worker interface

**Comment by stan-dot** [comment]
debugging the tests for the new feature now

**Comment by stan-dot** [comment]
image


@keithralphs what is the desired business logic for this kind of test?

**Review by callumforrester — CHANGES_REQUESTED** [review]
I would suggest an alternative implementation:

Replace `is_pending` and is `is_complete` on `Task` with a status enum and propagate that through the rest of the code. It simplifies your if check in the worker and also means you don't have to pass an ambiguous status string around.

**Comment by stan-dot** [comment]
ok this will take a while and a fair bit of logic too

**Comment by stan-dot** [comment]
there are two different models - classs TaskStatus in a worker but also TrackableTask.

this is a mess and too many classes.

**Comment by stan-dot** [comment]
workerEvent should just include a reference to that

**Comment by stan-dot** [comment]
you need to tell me the finite state machine logic of complete failed and error states for this @callumforrester .

**Comment by callumforrester** [comment]
@stan-dot the reason `WorkerEvent` doesn't reference the task directly is to handle the null case. Where the worker is not currently doing a task.

**Comment by callumforrester** [comment]
Another source of complexity is that the status of the worker and the status of the task the worker is currently doing are separate things and separately queryable. I would like it if it were simpler, I'm open to suggestions

File: src/blueapi/worker/reworker.py [hunk 1]
```diff
@@ -136,6 +136,21 @@ def submit_task(self, task: Task) -> str:
         self._tasks[task_id] = trackable_task
         return task_id
 
+    def get_tasks_by_status(self, status: str) -> list[TrackableTask[Task]]:
```
Comment (callumforrester) [inline src/blueapi/worker/reworker.py:139]: Status should be an enum

Comment (stan-dot) [reply src/blueapi/worker/reworker.py:139]: will do hectic refactoring tomorrow

Comment (stan-dot) [reply src/blueapi/worker/reworker.py:139]: update: changing the argus to be a required instance of the enum , without the big refactor

**Comment by stan-dot** [comment]
moved to #470 f

**Pull request closed without merging**